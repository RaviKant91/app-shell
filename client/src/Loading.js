/* 
 *  *************************************************
 *   BH [Highly] Confidential
 *   [Unpublished] Copyright 2020.  Baker Hughes
 *  
 *   NOTICE:  All information contained herein is, and remains the property of Baker Hughes, and/or
 *   its affiliates.  The intellectual and technical concepts contained herein are proprietary to Baker Hughes
 *   and/or its affiliates and may be covered by patents, copyrights, and/or trade secrets.  Dissemination of this information or
 *   reproduction of this material is strictly forbidden unless prior written permission is obtained from Baker Hughes.
 *  **************************************************
 *  
 */

import React from 'react';
import Grid from '@material-ui/core/Grid';
import withStyles  from '@material-ui/core/styles/withStyles';
import { BhSpinner } from './transformTagNameReactComponent';

const styles = theme => ({
  root: {display: 'flex', overflowY:'hidden', height:'99vh'},
  content: {flex: 1, height: '100vh', minHeight: 'calc(100vh - 8rem)'},
  progress: {margin: theme.spacing(1)}
});
  
const Loading = (props) => {
  const { classes } = props;
  return (
    <div className={classes.root} data-testid="loader">
    <Grid
      container
      direction="row"
      justify="center"
      alignItems="center"
    >
      <BhSpinner className={classes.progress} size={"xlarge"} position={"inline"} title="loader"/>
    </Grid> 
  </div>
  );
}

export default  withStyles(styles)(Loading);