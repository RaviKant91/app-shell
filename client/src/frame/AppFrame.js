/* 
 *  *************************************************
 *   BH [Highly] Confidential
 *   [Unpublished] Copyright 2020.  Baker Hughes
 *  
 *   NOTICE:  All information contained herein is, and remains the property of Baker Hughes, and/or
 *   its affiliates.  The intellectual and technical concepts contained herein are proprietary to Baker Hughes
 *   and/or its affiliates and may be covered by patents, copyrights, and/or trade secrets.  Dissemination of this information or
 *   reproduction of this material is strictly forbidden unless prior written permission is obtained from Baker Hughes.
 *  **************************************************
 *  
 */

import React from 'react';
import FullScreenNavigationDialog from './FullScreenNavigationDialog';
import FrameHolder from './FrameHolder';
import  Fetch  from '../network/Fetch';
import SnackBox from '../components/SnackBox';
import MessageContent from './MessageContent';
import isObject from 'lodash/isObject';
import DialogBox from '../components/DialogBox';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Button from '@material-ui/core/Button';
import { withTranslation } from 'react-i18next';
import ErrorOccurred from '../components/ErrorOccurred';

let ContextArray = [];
let eventPayloadObj = {};
let deletemyHistory;
export class AppFrame extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            url: this.props.url,
            mode:  this.props.mode,
            pageentry: this.props.pageentry,
            payload: this.props.payload,
            dialogShow:false,
            notitfication: {
                show:false,
                title:"Info",
                message: undefined,
                variant: 'info'
            },
            error: false
        }
        const nav = JSON.parse(window.localStorage.getItem('navigation')).find(o => o.id === this.props.appId);
        if(nav && nav != undefined){
            const event = new CustomEvent("onAppChange", {
                detail: nav,
                bubbles: true,
                cancelable: true
              });
              window.parent.document.dispatchEvent(event);
        }
    }
    handleCallback = () =>{
        this.setState({error: true})
    }
    handleErrorCallback = () =>{
        this.setState({error: false})
    }
    getUniqeContexts = (contArray, contextObj) =>{
        if(contArray.length > 0){
            contArray = contArray.filter(context => context.payLoad.eventName !== contextObj.payLoad.eventName);
        }
        contArray.push(contextObj);
        return contArray;
    }
    handleNotificationClose = (event, reason) => {
        if (reason === 'clickaway') {
          return;
        }
        this.setState({ 
            notitfication: {
                show:false,
                title:"Info",
                message: undefined,
                variant: 'info'
            } 
        });
        /* istanbul ignore if */
        if(this.state.notitfication.message === 'Session Timeout. ' ){
            this.pushToHistoryForLogout();
        }
      };
    proxyCall = (navObj) => {
            const options = {
                    url: navObj.pathname,
                    method: navObj.method,
                    data: navObj.state.detail, 
            }
            Fetch.post(options)
            .then(response=>response.json())
            .then(data=> {
                console.log(data); 
            }).catch(err => { 
                console.log(err);  
            });    
    }
    openNewTab = (navObj) => {  
        if(!navObj.pathname){
            navObj['pathname'] = (navObj.appname && navObj.appname != '') ? JSON.parse(window.localStorage.getItem('navigation')).find(o => o.id === navObj.appname).link:'';
        }
      var locationPath = window.location.hash ? `#${navObj.pathname}` : navObj.pathname,
        newTabWindow =  window.open(`${locationPath}`, '_blank');

        this.setContext(navObj);
        /* istanbul ignore else */
        if(navObj.state.detail !== undefined){
            //newTabWindow.contextPayload = navObj.state.detail;
            newTabWindow.contextPayload = window.getContext && window.getContext !== null? window.getContext(): navObj.state.detail;
            newTabWindow.getContext = () =>{ return window.getContext && window.getContext !== null? window.getContext(): navObj.state.detail;}
        }
    }
    pushToHistory = (navObj) => {
        //Context sharing add pathName if pathname not avaialble
        if(!navObj.pathname){
            navObj['pathname'] = (navObj.appname && navObj.appname != '') ? JSON.parse(window.localStorage.getItem('navigation')).find(o => o.id === navObj.appname).link:'';
        }
        if(navObj.hasOwnProperty("isUserDetailsUpdated")){
            navObj.pathname = localStorage.getItem("defaultAppLink");
        }
    this.setContext(navObj);
    /* istanbul ignore else */
       if(window.getContext){
            window.contextPayload = window.getContext();
       }      
        eventPayloadObj = navObj.state.detail;
        deletemyHistory = navObj.state.deletemyHistory?navObj.state.deletemyHistory:null;
        if(deletemyHistory && deletemyHistory!=undefined){
            this.props.history.replace(navObj);       
        }else{
            this.props.history.push(navObj);
        }
    }

    pushToHistoryForLogout = () => {
        if (window.localStorage || global.localStorage) {
            localStorage.removeItem("isAuthorized");
            localStorage.removeItem("activeToken");
            localStorage.removeItem("permissions");
            localStorage.removeItem("microapps");
            localStorage.removeItem("navigation");
            localStorage.removeItem("commonappImpacted");
            localStorage.removeItem("defaultAppLink");
            localStorage.removeItem("productName");
            localStorage.removeItem("app_link");
            localStorage.removeItem("authUrl");
            localStorage.removeItem("grantType");
            localStorage.removeItem("userInfoDialogMap");
            localStorage.removeItem("username");
            localStorage.removeItem("designTemplate");
            localStorage.removeItem("appfailed");
            localStorage.removeItem("headerMenuConfig");
            localStorage.removeItem("UserInfo");
            localStorage.removeItem("tenantid");
            localStorage.removeItem("tenantDropDown");
            localStorage.removeItem("searchConfig");
            localStorage.removeItem("menuItems");
            localStorage.removeItem("licencesUrl");
            localStorage.removeItem("thankYouMsg");
            localStorage.removeItem("unlicensedMessage");
            localStorage.removeItem("showHeaderMenuIcon");
            localStorage.removeItem("polling");
            localStorage.removeItem("wsport");
            localStorage.removeItem("selectedApp");
            localStorage.removeItem("displayMenuItemOnHeader");
            localStorage.removeItem("displayMicroappOnHeader");
            localStorage.removeItem("applicationName");
            localStorage.removeItem("advanceNotificationMeta");
            localStorage.removeItem("appVersion");
            window.location.reload();
          }
    }
    setContext = (navObj) => {
        window.getContext = null;
        if(navObj.hasOwnProperty("appname")){
             //context sharing add context in window context object
            const { detail } = navObj.state;
            //console.log('detail.payLoad.eventName',detail.payLoad.eventName)
            if(detail.payLoad && detail.payLoad.eventName){
                ContextArray = this.getUniqeContexts(ContextArray,detail);
            }
            window.getContext = () => {
                return ContextArray ;  
            };
            //  window.getContext = () => {
            //      return { context: { ...detail } };  
            //  };
        }else {
            const nav_url=navObj.pathname.replace(/[.*+#?^${}()|[\]\\]/g, "").substring(1);
            const app_name=nav_url.substring(0,nav_url.indexOf('/'));
            const add_url=nav_url.substring(nav_url.indexOf('/'));
            if(add_url != '/' && add_url != '')
            {
                navObj.pathname=`/${app_name}/`;
                navObj={...navObj, state:{...navObj['state'], detail:{...navObj['state']['detail']}}};
                window.getContext = () =>{ return [{ payLoad:{'route': add_url, ...navObj['state']['detail']}, detail:{...navObj['state']['detail']}}]};
            }
        }
     }
    openModal = (navObj) => {
        if(!navObj.pathname){
            navObj['pathname'] = (navObj.appname && navObj.appname != '') ? JSON.parse(window.localStorage.getItem('navigation')).find(o => o.id === navObj.appname).link:'';
        }
        this.setContext(navObj);
       if(window.getContext){
            window.contextPayload = window.getContext();
        }
        this.setState({
            url: navObj.pathname,
            mode:  navObj.state.mode,
            payload: navObj.state.detail,
        });
        
    }

    closeModal = () => {
        this.setState({
            url: this.props.url,
            mode: this.props.mode,
            payload: this.props.payload,
            pageentry:this.props.pageentry
        });
    }

    handleNavigation = (event) => {
        event.preventDefault();
        const navObj = isObject(event.data) ? JSON.parse( JSON.stringify(event.data) ) : JSON.parse( event.data );
        const { eventType, payload } = navObj;
        /* istanbul ignore else */
        if(eventType === 'navigation') {
            switch(payload.state.mode) {
                case "spa":
                    this.pushToHistory(payload);
                    break;      
                case "tab":
                    this.openNewTab(payload);
                     break;
                case "modal":
                     this.openModal(payload);
                     break;
                case "overlay":
                     this.openModal(payload);
                    break;
                default:
                     this.setState({
                         notitfication: {
                             show: true,
                             title:"Error",
                             variant: "error",
                             message: 'Invalid Navigation mode'
                         }
                     });     
            }
        } 
        else if(eventType === 'notification') {
                eventPayloadObj = "";
                this.setState({
                    notitfication: payload
                });
        }
        else if (eventType === 'proxy') {
            this.proxyCall(payload);
        }
        else if (eventType === 'close-overlay') {
            this.closeModal(payload);
        }
        else if (eventType === 'session missing') {
            localStorage.clear();
            this.props.history.push('/');
        }
        else if(eventType === 'session-timeout') {
            this.setState({
                eventType:eventType,
                dialogShow:true
            });
                localStorage.removeItem("isAuthorized");
                localStorage.removeItem("activeToken");
                localStorage.removeItem("permissions");
                localStorage.removeItem("microapps");
                localStorage.removeItem("navigation");
                localStorage.removeItem("commonappImpacted");
                localStorage.removeItem("defaultAppLink");
                localStorage.removeItem("productName");
                localStorage.removeItem("app_link");
                localStorage.removeItem("authUrl");
                localStorage.removeItem("grantType");
                localStorage.removeItem("userInfoDialogMap");
                localStorage.removeItem("username");
                localStorage.removeItem("designTemplate");
                localStorage.removeItem("appfailed");
                localStorage.removeItem("headerMenuConfig");
                localStorage.removeItem("tenantid");
                localStorage.removeItem("tenantDropDown");
                localStorage.removeItem("searchConfig");
                localStorage.removeItem("menuItems");
                localStorage.removeItem("licencesUrl");
                localStorage.removeItem("thankYouMsg");
                localStorage.removeItem("unlicensedMessage");
                localStorage.removeItem("advanceNotificationMeta");
                localStorage.removeItem("appVersion");

    }
      
    }  
    getlogouturl=()=>{
        fetch(`${window.location.pathname}logout`,{
           method: 'GET',
           headers: {
             'Accept': 'application/json',
             'Content-Type': 'application/json',
             'Cache-Control':' no-cache' 
           }
       })
       .then(response =>response.json())
       .then(data => {
        let logout = data && data.logout;
        let pathname = window.location.pathname;
        let windowOrg = window.hasOwnProperty( "location" ) && window.location.origin;
         localStorage.setItem("sessionLogout",true);
         localStorage.setItem('logoutSuccess',true)
         window.location.href = logout + windowOrg + (pathname === '/'?"":"/"+pathname.substring(1,pathname.length));
        
       }).catch(()=> this.setState({message: this.props.t('LoginError') ,loading:false}));
     }
    callTimeout=()=> {
        this.setState({dialogShow:false});
            localStorage.removeItem("isAuthorized");
            localStorage.removeItem("activeToken");
            localStorage.removeItem("permissions");
            localStorage.removeItem("microapps");
            localStorage.removeItem("navigation");
            localStorage.removeItem("commonappImpacted");
            localStorage.removeItem("defaultAppLink");
            localStorage.removeItem("productName");
            localStorage.removeItem("app_link");
            localStorage.removeItem("authUrl");
            localStorage.removeItem("grantType");
            localStorage.removeItem("userInfoDialogMap");
            localStorage.removeItem("username");
            localStorage.removeItem("designTemplate");
            localStorage.removeItem("appfailed");
            localStorage.removeItem("headerMenuConfig");
            localStorage.removeItem("sessionTimeOut");
            localStorage.removeItem("userInfo");
            localStorage.removeItem("tenantid");
            localStorage.removeItem("menuItems");
            localStorage.removeItem("licencesUrl");
            localStorage.removeItem("thankYouMsg");
            localStorage.removeItem("unlicensedMessage");
            localStorage.removeItem("advanceNotificationMeta");
            localStorage.removeItem("selectedTenant");
            localStorage.removeItem("appVersion");
            this.getlogouturl();
    }
     render() {
        const { mode , url, payload,dialogShow ,eventType, error} = this.state;  
        const {t}=this.props;
        const thirdPartyApp = this.props && this.props.thirdPartyApp;
        if(eventType=="session-timeout"){
            return(
                <DialogBox showDialog={dialogShow} id={"dialogPopup"}>
                    <DialogTitle id="alert-dialog-title">{t('SessionTimeout')}</DialogTitle>
                    <DialogContent>
                    <DialogContentText id="alert-dialog-description">
                    {t('SessionTimeoutInfo')}
                    </DialogContentText>
                    </DialogContent>
                    <DialogActions>
                    <Button  onClick={this.callTimeout} autoFocus>
                    {t('Close')}
                    </Button>
                    </DialogActions>
                </DialogBox>
            )
        }
        return (
            <React.Fragment>
            <SnackBox
                open={this.state.notitfication.show}
                autoHideDuration={6000}
                onClose={this.handleNotificationClose}
              >
                <MessageContent
                  onClose={this.handleNotificationClose}
                  title={this.state.notitfication.title}
                  variant= {this.state.notitfication.variant}
                  message= {this.state.notitfication.message}
                />
            </SnackBox>
               { mode === 'modal' ?
                        <FullScreenNavigationDialog {...this.props}   classes={{"closeButton":"abc"}} modalClose= {this.closeModal} >
                            <FrameHolder  url= {url} payload={payload} onFrameMessage = {this.handleNavigation} thirdPartyApp={thirdPartyApp} /> 
                     </FullScreenNavigationDialog>     : null   
               }
               {
                   mode=== 'overlay' ? <FrameHolder  url= {url} payload={payload} onFrameMessage = {this.handleNavigation} thirdPartyApp={thirdPartyApp} /> :null 

               }
               {error ? 
            <ErrorOccurred type={'load-failed'} app={"main"} thirdPartyApp={thirdPartyApp} parentCallback = {this.handleErrorCallback}/>
                    :<FrameHolder  parentCallback = {this.handleCallback} history={this.props.history} url= {this.props.url} payload={this.props.payload} eventObj={eventPayloadObj} onFrameMessage = {this.handleNavigation} pageentry={this.props.pageentry} thirdPartyApp={thirdPartyApp}/>}
            </React.Fragment> 
            )
    } 
}

export default withTranslation()(AppFrame);
