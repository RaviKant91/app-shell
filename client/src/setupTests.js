/* 
 *  *************************************************
 *   BH [Highly] Confidential
 *   [Unpublished] Copyright 2020.  Baker Hughes
 *  
 *   NOTICE:  All information contained herein is, and remains the property of Baker Hughes, and/or
 *   its affiliates.  The intellectual and technical concepts contained herein are proprietary to Baker Hughes
 *   and/or its affiliates and may be covered by patents, copyrights, and/or trade secrets.  Dissemination of this information or
 *   reproduction of this material is strictly forbidden unless prior written permission is obtained from Baker Hughes.
 *  **************************************************
 *  
 */

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// import Adapter from '@wojtekmaj/enzyme-adapter-react-17'; //TODO: upgrade to official version once available
configure({ adapter: new Adapter() });

const localStorageMock = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    clear: jest.fn()
  };
  global.localStorage = localStorageMock;
  jest.setTimeout(30000);