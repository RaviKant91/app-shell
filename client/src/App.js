/* 
 *  *************************************************
 *   BH [Highly] Confidential
 *   [Unpublished] Copyright 2020.  Baker Hughes
 *  
 *   NOTICE:  All information contained herein is, and remains the property of Baker Hughes, and/or
 *   its affiliates.  The intellectual and technical concepts contained herein are proprietary to Baker Hughes
 *   and/or its affiliates and may be covered by patents, copyrights, and/or trade secrets.  Dissemination of this information or
 *   reproduction of this material is strictly forbidden unless prior written permission is obtained from Baker Hughes.
 *  **************************************************
 *  
 */

import React, { Component ,lazy,Suspense} from 'react';
import { Switch ,Route   } from "react-router";
import { HashRouter,Redirect  } from 'react-router-dom';
import { PrivateRoute } from './PrivateRoute';
/*istanbul ignore next*/
 import RedirectTo from './RedirectTo';
/*istanbul ignore next*/
const LoginPage = lazy(() => import('./login/LoginPage'));
import Loading from './Loading';
import PreLoginPage from './login/PreLoginPage';
/*istanbul ignore next*/
import Auth from './Auth';
import i18N from './i18n'
import { I18nextProvider } from 'react-i18next';
let componentPromise,CustomHome;
import {setData} from './RbacData';
import PathParam from './login/PathParam';
export default class App extends Component {
  componentWillMount(){
    if(localStorage.getItem('designTemplate')){
      const filename=localStorage.getItem('designTemplate');
      componentPromise = import(`./templates/${filename}`);
      fetch(`${window.location.pathname}rbackData`, {
        method:"GET",
        headers: { 'Accept': 'application/json',
        'Content-Type': 'application/json' 
        }
      }).then(response => response.json())
      .then(data => { 
        setData(data.rbacData);
      });
    }
    else{
    //  fetch('/api/template', { method: 'GET', headers: { 'Accept': 'application/json', 'Content-Type': 'application/json','Cache-Control':' no-cache' } })
	     fetch(`${window.location.pathname}api/template`,  { method: 'GET', headers: { 'Accept': 'application/json', 'Content-Type': 'application/json' } })
      .then(response => response.json())
        .then(data => {
          localStorage.setItem('designTemplate', data.design_template);
          const filename=localStorage.getItem('designTemplate');
           componentPromise = import(`./templates/${filename}`);
           
        });
    }

  }
  render() {
    /*istanbul ignore next*/
    CustomHome = React.lazy(() => componentPromise);
    
    return (
       <Suspense fallback={<Loading/>}>
          <I18nextProvider i18n={i18N}>
      <HashRouter>
        <Switch>
          <Route exact path='/pathparam/:tenant' component={PathParam}/>
          <Route exact  path='/prelogin' render={(props) => {
            /*istanbul ignore next*/
        return (
          localStorage.getItem('isAuthorized')
          ? <Redirect to={{pathname: "/",  }}/> :<PreLoginPage {...props}/>
        )
       }} />
          <Route exact  path='/login' component={LoginPage} />
          <Route exact  path='/login/:tenant' component={LoginPage} />
          <Route exact path='/auth' component={Auth} />
          <PrivateRoute exact path='/redirecting' component={RedirectTo}/>
          
          <PrivateRoute path='/' component={CustomHome}/>
        </Switch> 
      </HashRouter>
      </I18nextProvider>
      </Suspense>
    );
  }
}