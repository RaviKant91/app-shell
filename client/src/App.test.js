/* 
 *  *************************************************
 *   BH [Highly] Confidential
 *   [Unpublished] Copyright 2020.  Baker Hughes
 *  
 *   NOTICE:  All information contained herein is, and remains the property of Baker Hughes, and/or
 *   its affiliates.  The intellectual and technical concepts contained herein are proprietary to Baker Hughes
 *   and/or its affiliates and may be covered by patents, copyrights, and/or trade secrets.  Dissemination of this information or
 *   reproduction of this material is strictly forbidden unless prior written permission is obtained from Baker Hughes.
 *  **************************************************
 *  
 */

import './setupTests';
import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory,createHashHistory } from 'history';
import { Switch ,Route   } from "react-router";
import { MemoryRouter } from "react-router-dom";
import { render,fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import {shallow,mount} from 'enzyme';
import App from './App';
import LoginPage from './login/LoginPage';
//import Home from './Home';
import RedirectTo from './RedirectTo';
import Loading from './Loading';
import AppFrame  from './frame/AppFrame';
import { wrap } from 'lodash';
const navigation=[
    {
        "id":"dashboard",
        "name": "Asset Dashboard",
        "link": "/asset-dashboard/",
        "icon": "speedrounded",
        "host": "https://ofe-dev-nginx-lb01-1933176829.us-west-2.elb.amazonaws.com",
        "path":"/asset-dashboard",
        "template":"/index.html",
        "navService":"/nav",
        "default": true
      },
      
      {
        "id":"appbuilder",
        "name": "App Builder",
        "link": "/app-builder/",
        "icon": "warningrounded",
        "host": "https://cde-sc-nginx-dev01-elb01-238862697.us-west-2.elb.amazonaws.com/appbuilder",
        "path":"/app-builder",
        "template":"/index.html",
        "navService":"/nav",
        "default": false
      }

];
jest.mock("react-i18next", () => ({
  withTranslation: () => (Component) => {
      Component.defaultProps = {...Component.defaultProps, t: (key) => key};
      return Component;
    },
}));
describe('<App/>',()=>{
    it('renders properly',()=>{
        window.localStorage.setItem('designTemplate','default');
        const wrapper=shallow(<App />);
        expect(wrapper).toMatchSnapshot();

    });
    it('without design Template render',()=>{
      window.localStorage.removeItem('designTemplate');
      const mockComponentWillMountResponse = {
                responseStatus:200,
                message:'success',
                design_template:"default"
           };
            const mockFetchPromise = Promise.resolve({ 
              json: () => mockComponentWillMountResponse,
            });
            jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise); 
      const wrapper=shallow(<App />);
      expect(wrapper).toMatchSnapshot();
    });
	//  it('verify template API called',(done) =>{
  //     const event={
  //         preventDefault:()=>{}
  //     }
  //     const mockComponentWillMountResponse = {
  //         responseStatus:200,
  //         message:'success',
  //         design_template:"default"
  //    };
  //     const mockFetchPromise = Promise.resolve({ 
  //       json: () => mockComponentWillMountResponse,
  //     });
  //     jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise); 
  //     let wrapper=shallow(<App/>);
  //      process.nextTick(() => {
  //         expect(global.fetch).toHaveBeenCalled();
  //         expect(global.fetch).toHaveBeenCalledWith('/api/template', {
  //             method: 'GET',
  //             headers: 
  //             {
  //               'Accept': 'application/json',
  //               'Content-Type': 'application/json'
  //             }
  //           });   
  //           expect(window.localStorage.getItem('designTemplate')).toBe("default");
  //         global.fetch.mockClear();
  //         done();
  //       }); 
  //   })

    it('routes /login to LoginPage', () => {
        global.fetch = jest.fn(() =>
          Promise.resolve({
            json: () => Promise.resolve({"data": "data"}),
            status:200,
            ok: true
          })
        );
        const wrapper = shallow(<App />);
        expect(wrapper.find('Route').first().props().path).toBe('/pathparam/:tenant');
      });

    it('routes /redirecting to RedirectTo', () => {
      let wrapper = shallow(
        <MemoryRouter initialEntries={['/redirecting']}>
          <Switch>
            
            <Route path='/redirecting' component={RedirectTo} />
          
          </Switch>
        </MemoryRouter>
      );
      expect(wrapper.find("Route").props().path).toBe('/redirecting');
      expect(wrapper.find("Route").props().component).toBe(RedirectTo);
      
    });


      // it('login page load', () => {
      //   const mockSuccessResponse = {
      //     responseStatus:200,
      //     message:'success',
      //   };
      //   const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
      //   const mockFetchPromise = Promise.resolve({ // 3
      //     json: () => mockJsonPromise,
      //   });
      //   jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise); 
      //   let wrapper = mount(
      //     <MemoryRouter initialEntries={['/login']}>
      //       <Switch>
             
      //         <Route path='/login' component={LoginPage} />
           
      //       </Switch>
      //     </MemoryRouter>
      //   )
      //   expect(wrapper.find(LoginPage)).toHaveLength(1);
      // });

     

      // it('Redirect to laoding invoked on submit or not', () => {
      //   let wrapper = mount(
      //     <MemoryRouter initialEntries={['/redirecting']}>
      //       <Switch>
             
      //         <Route path='/redirecting' component={RedirectTo} />
           
      //       </Switch>
      //     </MemoryRouter>
      //   );
      //  expect( wrapper.find(Loading)).toHaveLength(1);
      //  const handlFormSubmit=jest.fn();
      // });

});

const loginsetup = () => {
  const history = createMemoryHistory()
  const route = '/login'
  history.push(route)
  const { getByTestId } = render(
      <Router history={history}>
        <LoginPage />
      </Router>
    )
  const emailInput = getByTestId('nameInput');
  const passwordInput = getByTestId('passwordInput');
  const submit = getByTestId('submit');
  
  return {
      emailInput,
      passwordInput,
      submit
  }
}

describe('<Login/>',()=>{

  // test('rendering a component that uses withRouter', () => {
  //   const mockSuccessResponse = {
  //     responseStatus:200,
  //     message:'success',
  //   };
  //   const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
  //   const mockFetchPromise = Promise.resolve({ // 3
  //     json: () => mockJsonPromise,
  //   });
  //   jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise); 
  //     const history = createMemoryHistory()
  //     const route = '/login'
  //     history.push(route)
  //     const { getByTestId } = render(
  //       <Router history={history}>
  //         <LoginPage />
  //       </Router>
  //     )
  //     expect(getByTestId('login-container')).toBeInTheDocument();   
  
  //   });

    // test('rendering a component that uses withRouter', () => {
    //   const mockSuccessResponse = {
    //     responseStatus:200,
    //     message:'success',
    //   };
    //   const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
    //   const mockFetchPromise = Promise.resolve({ // 3
    //     json: () => mockJsonPromise,
    //   });
    //   jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise); 
    //   const { emailInput,passwordInput } = loginsetup();
    //   expect(emailInput).toHaveValue('');
    //   expect(passwordInput).toHaveValue('');
    // });

    // test('username check',()=>
    // {
    //   const mockSuccessResponse = {
    //     responseStatus:200,
    //     message:'success',
    //   };
    //   const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
    //   const mockFetchPromise = Promise.resolve({ // 3
    //     json: () => mockJsonPromise,
    //   });
    //   jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise); 
    //     const { emailInput } = loginsetup();
    //     fireEvent.change(emailInput, { target: { value: 'rukmani' } });
    //     expect(emailInput.value).toBe('rukmani');
    
    // });

    // it('password check',()=>{
    //   const mockSuccessResponse = {
    //     responseStatus:200,
    //     message:'success',
    //   };
    //   const mockJsonPromise = Promise.resolve(mockSuccessResponse); // 2
    //   const mockFetchPromise = Promise.resolve({ // 3
    //     json: () => mockJsonPromise,
    //   });
    //   jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise); 
    //   const { passwordInput } = loginsetup();
    //   fireEvent.change(passwordInput, { target: { value: 'rukmani123' } });
    //   expect(passwordInput.value).toEqual('rukmani123');
    //   });
  
});

describe('<RedirectTo/>',()=>{

  // test('App bar rendered',async  () => {
  
  //     const history = createMemoryHistory()
  //     const path = `/redirecting`;
  //     localStorage.setItem('isAuthorized',true);
  //     localStorage.setItem('navigation',JSON.stringify(navigation));
    
  //     history.push(path)
  //     console.log(history);
  //     const { getByTestId } = render(
  //       <Router history={history}>
  //         <RedirectTo />
  //       </Router>
  //     )
   
  //     expect(getByTestId('loader')).toBeInTheDocument();   
  
  //   });
  
});

 /** test Home */
const homesetup=()=>{
  const history = createHashHistory()
  const route = '"#/app-builder/'
  history.push(route);
  history.location.pathname='/app-builder/';
  history.location.state=''; 
  const { getByTestId } = render(    
<Router history={history} location={history.location}>
{/* <Home/> */}
</Router>
    
)
  return getByTestId;
};

describe('<Home/>',()=>{

      // test('Apps Bar Rendered',()=>
      // {
      //   window.localStorage.setItem('navigation',JSON.stringify(navigation));
      //   const getByTestId=homesetup();
      //   expect(getByTestId('app-bar')).toBeInTheDocument();   
      // });

      // test('Apps List Rendered',()=>
      // {
      //   window.localStorage.setItem('navigation',JSON.stringify(navigation));
      //   const getByTestId=homesetup();
      //   expect(getByTestId('app-list')).toBeInTheDocument();  
    
      // });
 
      // test('Apps List child Rendered',()=>
      // {
      //   window.localStorage.setItem('navigation',JSON.stringify(navigation));
      //   const getByTestId=homesetup();
      //   const appList = getByTestId('app-list');
      //   expect(getByTestId('app-list')).toBeInTheDocument();  
      //   expect(appList.children.length).toBe(2); 
      // });


  });


  describe('App frame component inital state',()=>{
    const menuItems =[{
      'id':'cases',
      'name':'case',
      "displayTextId":"cases",
      'default':'dashboard'
    }]
    // it('verify notification message state',()=>{
    //   window.localStorage.setItem('navigation',JSON.stringify(navigation));
    //     window.localStorage.setItem('menuItems',JSON.stringify(menuItems));
    //     const component = shallow(<AppFrame url="/asset-dashboard/" mode="spa" pageentry="" payload="{}" />);
    //     const instance = component.instance();
    //     expect(component.state().notitfication.message).toBeUndefined();
    // });
  
});






