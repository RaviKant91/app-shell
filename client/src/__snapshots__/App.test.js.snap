// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders properly 1`] = `ShallowWrapper {}`;

exports[`<App/> without design Template render 1`] = `ShallowWrapper {}`;
